smithy-aws-apigateway-openapi/src/main/java/software/amazon/smithy/aws/apigateway/openapi/AddIntegrations.java

```diff
@@ -121,6 +122,21 @@ private static ObjectNode getIntegrationAsObject(
         return integrationNode;
     }
 
+    private static void validateTraitConfiguration(IntegrationTrait trait,
+                                                   Context<? extends Trait> context,
+                                                   OperationShape operation) {
+        // For HTTP APIs, API Gateway requires that the payloadFormatVersion is set on integrations.
+        // https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-integration.html
+        // If the payloadFormatVersion has not been set on an integration and the apiGatewayType has been set to "HTTP",
+        // the conversion fails.
+        ApiGatewayConfig.ApiType apiType = context.getConfig().getExtensions(ApiGatewayConfig.class)
+                .getApiGatewayType();
+        if (!trait.getPayloadFormatVersion().isPresent() && apiType.equals(ApiGatewayConfig.ApiType.HTTP)) {
+            throw new OpenApiException("When using the HTTP apiGatewayType, a payloadFormatVersion must be set on the"
```

Comment:
I think this should be clarified a little. It's important here to show the source of the conflict and guide to where this should be fixed.

```
String.format("When the 'apiGatewayType' OpenAPI conversion setting is 'HTTP', a 'payloadFormatVersion' must "
        + "be set on the aws.apigateway#integration trait applied at %s", trait.getSourceLocation());
```

Thought I submitted this in the first PR, apologies.

Reply:
I updated the error message in 9c1f284, but did not include the sourceLocation of the trait.  That sourceLocation points back to where the `aws.api#integration` trait itself was defined, not where it was applied on the service or operation, so it wouldn't be helpful in this case.